packages/x-components/src/x-modules/search/components/redirection.vue

```diff
@@ -67,7 +71,7 @@
      * @internal
      */
     protected get redirection(): RedirectionModel | null {
-      return this.redirections[0] ?? null;
+      return this.redirections?.[0] ?? null;
```

Comment:
Redirections are right now required, so you don't need this operator. Previously I found an issue where because of using `any` as type, they were not generated for a mock. This issue has been fixed now.

I would like to make them optional (together with the promoted and banners) in the `SearchResponse`, as making them optional will force us to properly handle cases like the one we have here, but that's another story.

Reply:
I see a lot of errors in e2e tests, because of this. So maybe we have to change the types, but at the moment I will keep this, to avoid those errors.